### ISSUE TITLE
Fail fast for overwrite-related errors in cellpose/napari-workflows tasks

### ISSUE THREAD

**Issue by tcompa** [fractal-analytics-platform/fractal-tasks-core#505]
(branching from #491 and #499, since this issue has lower priority)

For cellpose/napari-workflows, we should try to fail as early as possible if `overwrite=False` and some output already exists.

For instance we could:

1. Verify that `prepare_label_group` is always called _before_ actual processing.
2. Perform some parts of the `write_table` checks before actual processing. Should we split `write_table` into `prepare_table_group` and `write_table`? I'm not sure yet that the additional complexity is worth it.. Maybe we can simply have a `preliminary_check_for_write_table` (random name..) function, and keep `write_table` as is. TBD.

**Comment by jluethi** [comment]
Good point. I'd prefer the `preliminary_check_for_write_table` option over splitting `write_table` :)
